extensions/amp-story/1.0/audio.js

```diff
@@ -34,3 +34,16 @@ export function upgradeBackgroundAudio(element, loop = true) {
 
   return audioEl;
 }
+
+/**
+ * Waits for elements that can resolve their audio status on runtime.
+ * @param {!Element} element the root
+ * @return {!Promise}
+ */
+export function waitForElementsWithUnresolvedAudio(element) {
+  return Promise.all(
+    Array.from(element.querySelectorAll('amp-video[cache]:not([noaudio])')).map(
+      (el) => el.getImpl()
```

Comment:
How does this guarantee that the video cache XHR came back? You'd need to at least wait for `buildCallback` to resolve, using signals

Reply:
Doesn't `getImpl()` wait for `buildCallback`? The implementation has a default parameter `waitForBuild = true`: https://github.com/ampproject/amphtml/blob/main/src/custom-element.js#L1561